cloud/amazon/ec2_vpc_nacl.py

```diff
@@ -0,0 +1,1472 @@
+        if self.nacl_identifier and \
+                not is_cidr_format(self.nacl_identifier) and \
+                not resembles_nacl_id(self.nacl_identifier) and \
+                'Name' not in tags:
+            tags.update({'Name':  self.nacl_identifier})
+
+        if target_nacl and tags:
+            if target_nacl.tags != tags:
+                target_nacl.add_tags(tags)
+                actionresult.changed = True
+
+        return actionresult
+
+    def modify_nacl_to_ruleset_action(self, actionresult):
+        '''
+        Modify the current network ACL to match the current ruleset.
+        '''
+        if not actionresult.nacl:
+            return actionresult
+
+        if not self.ruleset_present:
+            return actionresult
+
+        current_ruleset = RuleSet(actionresult.nacl)
+        target_ruleset = RuleSet(self.ruleset)
+
+        to_be_created = target_ruleset.difference(current_ruleset)
+        to_be_removed = current_ruleset.difference(target_ruleset)
+
+        is_egress = lambda r: unicode(r.egress).lower() == 'true'
+
+        for rule in to_be_removed:
+            self.aws.delete_networkacl_entry(
+                actionresult.nacl.id, rule.rule_number, is_egress(rule))
+            actionresult.removed_rules.append(rule)
+            actionresult.changed = True
+
+        for rule in to_be_created:
+            self.aws.create_network_acl_entry(
+                actionresult.nacl.id, **rule.as_dict())
+            actionresult.added_rules.append(rule)
+            actionresult.changed = True
+
+        return actionresult
+
+
+#  exceptions #################################################################
+class LocalModuleException(Exception):
+    pass
+
+
+class TooManyItems(LocalModuleException):
+    pass
+
+
+#  main #######################################################################
+def get_ansible_module():
+    valid_states = list(chain(*NetworkACLModule.STATES.values()))
+
+    argument_spec = ec2_argument_spec()
+
+    argument_spec.update(dict(
+        egress=dict(
+            required=False, type='list', default=list(),
+            aliases=['rules_egress']),
+        ingress=dict(
+            required=False, type='list', default=list(),
+            aliases=['rules']),
+        nacl=dict(required=False, aliases=['name']),
+        purge_rules=dict(required=False, type='bool', default=False),
+        state=dict(default='present', choices=valid_states),
+        subnets=dict(required=False, type='list', default=list()),
+        tags=dict(required=False, type='dict', aliases=['resource_tags']),
+        use_default_nacl=dict(required=False, type='bool', default=False),
+        vpc=dict(required=False, aliases=['vpc_id']),
+        ),
+    )
+
+    return AnsibleModule(argument_spec=argument_spec,
+                         mutually_exclusive=[['nacl', 'use_default_nacl']])
+
+
+def main():
+
+    module = get_ansible_module()
+    results = ActionResults()
+
+    try:
```

Comment:
we shy away from generic/catch all exception handling because it just shows the underlying library messages, most are very unhelpful. We try to prep-end a useful message using the specific context of the action the module was trying to engage in, possibly pointing at problems/solutions.